src/java/org/apache/cassandra/io/sstable/compaction/SortedStringTableCursor.java

```diff
@@ -97,6 +108,33 @@ public SortedStringTableCursor(SSTableReader sstable, RandomAccessReader dataFil
             this.regularColumns = toArray(header.columns(false));
             this.staticColumns = toArray(header.columns(true));
             this.columnsReusableArray = new ColumnMetadata[Math.max(regularColumns.length, staticColumns.length)];
+
+            if (tokenRange != null && !tokenRange.right.isMinimum() && tokenRange.right.compareTo(sstable.last.getToken()) < 0)
+            {
+                final RowIndexEntry position = sstable.getPosition(tokenRange.right.maxKeyBound(), SSTableReader.Operator.GT);
+                if (position != null)
+                    this.endPosition = position.position; // 0 if end is before our first key.
+                else
+                {
+                    assert false : "Range " + tokenRange + " end is before last sstable token " + sstable.last.getToken() + " but no position was found";
```

Comment:
Why is this only `assert false` as opposed to always throwing an `IllegalArgumentException`?

Reply:
Changed to `AssertionError`, because this is a case of broken sstable invariants rather than illegal input.

Reply:
Thanks for the explanation.